### PR TITLE
[Version] 1.0.1 fix and improve typings, improve .npmigonre

### PR DESCRIPTION
**Pull request by Kazimirkas**

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+.vscode
+src
+.editorconfig
+.gitignore
+.prettierrc
+global.d.ts
+rollup.config.js
+tsconfig.json
+tsconfig.rules.json
+tslint.json
```
Comment (RomiC) [inline .npmignore]: don't forget `examples`

File: .npmignore [hunk 2]
```diff
@@ -0,0 +1,11 @@
+__tests__
+.vscode
```
Comment (RomiC) [inline .npmignore]: From my point of view, it's better to remove this file from repo completely, if it doesn't contain any specific settings, which are necessary for working on a project.

File: .npmignore (hunk 1 above)
Comment (Kazimirkas) [reply .npmignore]: thanks

File: .npmignore (hunk 2 above)
Comment (Kazimirkas) [reply .npmignore]: .vscode is added to .gitignore and need to be ignored here as well

**Comment by RomiC** [comment]
@Kazimirkas how about to completely remove the `/examples` folder. I'm not sure if it's useful at all

**Comment by Kazimirkas** [comment]
+

**Pull request merged**